app-backend/batch/src/main/scala/com/azavea/rf/batch/airflow/export/CreateExportDef.scala

```diff
@@ -0,0 +1,165 @@
+      try {
+        (r53Client
+          .listResourceRecordSets(new ListResourceRecordSetsRequest(r53id))
+          .getResourceRecordSets()
+          .asScala.toList
+          .filter(r => r.getName() == "dataproc.rasterfoundry.com.")
+          .head
+        )
+      } catch {
+        case e:Exception => {
+          logger.error(s"Could not retrieve resource record sets due to this error:\n${e.getStackTraceString})")
+          throw new Exception("Could not retrieve resource record sets")
+        }
+      } finally {
+        logger.info("Retrieved resource record sets")
+      })
+
+    val clusterId = (
+      try {
+        resourceRecordSets.getResourceRecords().asScala.toList.head.getValue().replace("\"", "")
+      } catch {
+        case e:Exception => {
+          logger.error(s"Could not get Spark cluster ID due to this error:\n${e.getStackTraceString})")
+          throw new Exception("Could not get Spark cluster")
+        }
+      })
+    
+    clusterId
+  }
+
+  protected def startExportEmrJob(exportDef: ExportDefinition, exportDefUri: String): Unit = {
+    val jarStep = new HadoopJarStepConfig()
+    jarStep.setJar(jarPath)
+    jarStep.setArgs(List("/usr/bin/spark-submit", "--master", "yarn", "--deploy-mode",
+          "cluster", "--class", "com.azavea.rf.batch.export.Export", "--driver-memory", "2G",
+          "s3://rasterfoundry-global-artifacts-us-east-1/batch/rf-batch-b933688.jar",
+          "-j", s"s3://${exportDefUri}/${exportDef.id}.json").asJava)
+
+    val steps = new StepConfig()
+    steps.setName(s"export-${exportDef.id}")
+    steps.setActionOnFailure("CONTINUE")
+    steps.setHadoopJarStep(jarStep)
+
+    val jobSteps = new AddJobFlowStepsRequest()
+    jobSteps.setJobFlowId(getClusterId())
+    jobSteps.setSteps(List(steps).asJava)
+
+    val awsCreds = new DefaultAWSCredentialsProviderChain().getCredentials()
+
+    val emrClient = new AmazonElasticMapReduceClient(awsCreds)
+    emrClient.addJobFlowSteps(jobSteps)
+    ()
+  }
+
+  def run: Unit = {
+    logger.info("Starting export process...")
+
+    val createExportDef = for {
+      user: User <- OptionT(Users.getUserById(airflowUser))
+      export: Export <- OptionT(database.db.run(Exports.getExport(exportId, user)))
+      exportDef: ExportDefinition <- OptionT(Exports.getExportDefinition(export, user))
+    } yield {
+      val exportDefUri = writeExportDefToS3(exportDef)
+      logger.info(s"Export definition ${exportDef.id} put to S3")
+      startExportEmrJob(exportDef, exportDefUri)
+    }
+    
+    val result = Await.ready(createExportDef.value, Duration.Inf)
+    result onComplete {
+        case Success(result) => {
+          result match {
+            case Some(_) => {
+              logger.info("Export completed successfully")
+              stop
+            }
+            case _ => {
+              logger.info("Export failed")
+            }
+          }
+          stop
+        }
+        case Failure(e) => {
+          stop
```

Comment:
actor system termination should be done after `sendError(e)`, otherwise there would be an exception (smth like trying to create actor in a stopped system).